godot/src/ui/components/map_satellite/map_component.gd

```diff
@@ -0,0 +1,323 @@
+class_name MapComponent
+extends Control
+
+signal clicked_parcel(parcel: Vector2i)
+
+const IMAGE_FOLDER = "res://src/ui/components/map_satellite/assets/4/"
+const TILE_SIZE = Vector2(512, 512)
+const GRID_SIZE = Vector2(16, 16)
+const PARCELS_PER_TILE = Vector2(20, 20)
+const PARCEL_SIZE = TILE_SIZE / PARCELS_PER_TILE
+const MIN_ZOOM := Vector2(.5, .5)
+const MAX_ZOOM := Vector2(2, 2)
+const TAP_THRESHOLD := 10.0  # en píxeles
+const MAP_SIZE = PARCEL_SIZE * Vector2(340, 340)
+const PARCEL_OFFSET = Vector2i(170, 170)
+const MAP_CENTER = MAP_SIZE / 2
+const TILE_DISPLACEMENT = Vector2(18, 18) * PARCEL_SIZE
+const MAP_MARKER = preload("res://src/ui/components/map_satellite/map_marker.tscn")
+const MAP_PIN := preload("res://src/ui/components/map_satellite/map_pin.tscn")
+const ARCHIPELAGO_CIRCLE = preload("res://src/ui/components/map_satellite/archipelago_circle.tscn")
+
+var dragging := false
+var touch_start_pos := Vector2.ZERO
+var touch_id := -1
+var last_pinch_distance := 0.0
+var active_touches := {}
+var just_zoomed := false
+var poi_places_ids = []
+
+@onready var camera := $Camera2D
+@onready var map_marker: Marker = %MapMarker
```

Comment:
Can you start the map_marker invisible? and is made it visible only when it is used